### ISSUE TITLE
[request] add packages for apps section in awesome-remarkable

### ISSUE THREAD

**Issue by raisjnn** [matteodelabre/toltec#9]
https://github.com/rehackable/awesome-remarkable

some missing packages:

* keywriter
* wikipedia
* rmvncclient
* pocket2rm

non listed packages that might be interesting:

* recrossable
* syncthing
* ???

**Comment by Eeems** [comment]
- [ ] [Crazy Cow](https://github.com/machinelevel/sp425-crazy-cow) (MIT)
- [x] [rmWacomToMouse](https://github.com/LinusCDE/rmWacomToMouse) (MIT)
- [ ] [srvfb](https://github.com/merovius/srvfb) (Apache 2.0)
- [ ] [remarkable_news](https://github.com/evidlo/remarkable_news) (GPL-3.0)
- [ ] [remarkable-touchgestures](https://github.com/ddvk/remarkable-touchgestures) (MIT)
- [ ] [reStream](https://github.com/rien/reStream) (MIT)
- [x] [remarkable-splash](https://github.com/ddvk/remarkable-splash) (MIT)
- [ ] [1bitpaint](https://github.com/HiroshiTamura/1bitpaint_for_reMarkable) (MIT)
- [ ] [ReMarkable.NET](https://github.com/parzivail/ReMarkable.NET) (MIT)
- [ ] [Xochitl templates](https://remarkablewiki.com/tips/templates)
    - https://github.com/bliekp/remarkable-templates (CC0-1.0)
    - https://github.com/oisinmulvihill/reMarkableTemplates (GPL-3.0)

**Comment by matteodelabre** [comment]
Thanks for the suggestions. Some of these apps lack a license that would allow us to redistribute it, I’ve marked them by editing your comments.

**Comment by Eeems** [comment]
Time to open up issues asking for a license to be added :)

**Edit:** I've opened issues and edited the comment with the issues.

**Comment by matteodelabre** [comment]
Thanks a lot @Eeems!

**Comment by Eeems** [comment]
Looks like srvfb does have a license (Apache 2.0) at the bottom of the README, and in the header of every file.

**Comment by LinusCDE** [comment]
- [x] rmWacomToMouse / rmServeWacomInput

I packaged [rmWacomToMouse](https://github.com/LinusCDE/rmWacomToMouse) yesterday. I named it after the server side component (the c version) rmServeWacomInput (as a package in all lowercase). I still need to update the readme though.

Can be run as just the command or with an [mock frontend](https://github.com/LinusCDE/appmarkable) also in launchers ([Image](https://user-images.githubusercontent.com/22298664/92416846-b8a17e80-f15f-11ea-8391-78765633cf92.jpg)).

Not sure if i should change the name of the app to the repo name though. It has its pros and cons.

**Comment by LinusCDE** [comment]
Are remarkable_{mouse,keyboard} not host-only?

If I'm not mistaken they just ssh into the device and cat the the event device out.

There is really is not onboard software to package.

Although maybe one could add a package that autoinstalls the pdf into xochitls files. I think I have such a script lying around somewhere.

**Comment by Eeems** [comment]
> Are remarkable_{mouse,keyboard} not host-only?
> 
> If I'm not mistaken they just ssh into the device and cat the the event device out.
> 
> There is really is not onboard software to package.
> 
> Although maybe one could add a package that autoinstalls the pdf into xochitls files. I think I have such a script lying around somewhere.

@LinusCDE Good point, I didn't look too closely and it seemed kind of like there was a UI on the keyboard one. I've crossed them off.

**Comment by LinusCDE** [comment]
@Eeems Technicially I could just take the picture and use it as a background with [appmarkable](https://github.com/LinusCDE/appmarkable) while running something stupid like sleep. That would techincically be an app then which may be easier for users.

However that is only an idea that @Evidlo would have to pick up if he wanted.

**Comment by Eeems** [comment]
#41  will handle koreader.

**Comment by Eeems** [comment]
Another suggestion rm.NET

**Comment by Eeems** [comment]
Some more suggestions:
- [1bitpaint](https://github.com/HiroshiTamura/1bitpaint_for_reMarkable) (MIT)
- [recrossable](https://github.com/sandsmark/recrossable) (https://github.com/sandsmark/recrossable/issues/2)

**Comment by matteodelabre** [comment]
- [ ] [Xochitl templates](https://remarkablewiki.com/tips/templates)
    - https://github.com/bliekp/remarkable-templates (CC0-1.0)
    - https://github.com/oisinmulvihill/reMarkableTemplates (GPL-3.0)

**Comment by Eeems** [comment]
- [ ] [remarkable-splash](https://github.com/ddvk/remarkable-splash) (MIT)

**Comment by Eeems** [comment]
Thoughts on adding lz4 so that getting it installed and kept up to date for reStream is easier? https://github.com/rien/reStream#sub-second-latency

**Comment by Eeems** [comment]
It looks like the micro editor will work OOB, we should probably package it up: https://micro-editor.github.io/

**Comment by matteodelabre** [comment]
> Thoughts on adding lz4 so that getting it installed and kept up to date for reStream is easier? https://github.com/rien/reStream#sub-second-latency

lz4 should already be in Entware according to https://github.com/Entware/Entware/issues/556, but I don’t see it in their published repositories for some reason.

**Comment by LinusCDE** [comment]
> > Thoughts on adding lz4 so that getting it installed and kept up to date for reStream is easier? https://github.com/rien/reStream#sub-second-latency
> 
> lz4 should already be in Entware according to [Entware/Entware#556](https://github.com/Entware/Entware/issues/556), but I don’t see it in their published repositories for some reason.

Sounds like a nice package to add. reStream specifically now used the approach to create a rust based app that included lz4 statically. It was recently [added as a package](https://github.com/toltec-dev/toltec/tree/testing/package/restream) in PR #184 .

**Comment by matteodelabre** [comment]
I missed that addition, thanks for the link! At a glance it seems that the Rust app uses the `lz-fear` crate, which actually re-implements lz4 in pure Rust (in contrast to the `lz4` crate). If I understand correctly, neither the lz4 C library nor the lz4 binaries are needed anymore, unless someone knows of another rM project that uses it?

**Comment by Eeems** [comment]
Not that I'm aware of. And likely the package will eventually drop in entware.

**Comment by Eeems** [comment]
https://github.com/ddvk/remarkable-stylus

**Comment by Eeems** [comment]
uinput kernel module

**Comment by ozel** [comment]
python3 (with functional pip) and/or CLI tools that allow folder syncing via webdav/dropbox etc. on the tablet

**Comment by Eeems** [comment]
> python3 (with functional pip) and/or CLI tools that allow folder syncing via webdav/dropbox etc. on the tablet

Python is already available through entware: https://bin.entware.net/armv7sf-k3.2/

**Comment by LinusCDE** [comment]
> python3 (with functional pip) and/or CLI tools that allow folder syncing via webdav/dropbox etc. on the tablet

Those are already provided by entwares repo.

**Comment by ozel** [comment]
ups, how embarrassing! thanks for pointing it out. This invalidates my entire request of course. Somehow I had assumed only the packages listed on the toltec page would be part of this...

**Comment by matteodelabre** [comment]
The Entware packages are not currently listed, this feature is tracked in #139.

**Comment by Eeems** [comment]
Thoughts on adding gocryptfs?

https://github.com/plan5/remarkable-gocryptfs is instructing users to manually download and compile it on device, we probably could lower the barrier to entry there. @plan5 this might be a good first package for you ;)

**Comment by plan5** [comment]
Maybe we should add fusemodule as a package request, too. It's needed for gocryptfs on the rM1 (rM2 has it built in). I'm still stuck trying to build it.

**Comment by plan5** [comment]
I'm finalizing the recipe for [innernet](https://github.com/tonarino/innernet) right now. It's a fancy wrapper for wireguard, I would say. Makes configuration much easier. I came across it in the Arch community repo first.
PR incoming :)

**Comment by Eeems** [comment]
You beat me to this :P

**Comment by koenvervloesem** [comment]
Would [reMarkable Calendar Creator](https://github.com/koenvervloesem/remarkable-calendar-creator) be interesting to add? I [shared it recently on Reddit](https://www.reddit.com/r/RemarkableTablet/comments/n7whhe/remarkable_calendar_creator_generates_a_new/) and it was well received. I've fixed some minor issues now after that feedback and I'm finally comfortable with sharing it more widely. I can tag a release and open a PR for a Toltec package if there's enough interest. All dependencies are already in Toltec.

**Comment by danshick** [comment]
@koenvervloesem Sure, feel free to package it and submit a PR. If you need any help, we'd be happy to offer advice.

**Comment by Sami32** [comment]
My ebooks reader package suggestion is the following:
https://github.com/edrlab/thorium-reader

It's by far the most ePUB 3.x compliant reader including CSS3, HTML5 and MathML if we doesn't consider the Calibre E-book viewer bundle not adequat for such device type.
KOreader have no real ePUB 3.x support and doesn't seem to have it on its roadmap.

**Comment by Eeems** [comment]
It looks like this would first require doing things like porting electron.

**Comment by Sami32** [comment]
@Eeems Could this help?

https://github.com/vulcanoio/meta-electron
https://web.archive.org/web/20210122193557/https://developer.toradex.com/knowledge-base/electron-framework

**Comment by Sami32** [comment]
rclone
https://remarkablewiki.com/tips/rclone
https://rclone.org
https://github.com/rclone/rclone

Bypassing the 8 GB Remarkable proprietary Cloud limitation should be welcome IMHO.

**Comment by Eeems** [comment]
> @Eeems Could this help?
> 
> https://github.com/vulcanoio/meta-electron
> https://web.archive.org/web/20210122193557/https://developer.toradex.com/knowledge-base/electron-framework

Porting electron also would require handling a windowing system as we don't currently have X or wayland running. Not really something I think we are interested in dealing with as part of toltec.



> rclone
> https://remarkablewiki.com/tips/rclone
> https://rclone.org
> https://github.com/rclone/rclone
> 
> Bypassing the 8 GB Remarkable proprietary Cloud limitation should be welcome IMHO.

rclone is already available as part of entware. Just run `opkg install rclone`.

**Comment by Sami32** [comment]
@Eeems I see.
Thank you for your feedback +1

**Comment by Eeems** [comment]
We should probably look into including some font packages at some point.

**Comment by gustaphe** [comment]
`xkbmap` is needed as a compliment to `keyd`, for people who use the Scandinavian keyboard.

**Comment by Eeems** [comment]
> `xkbmap` is needed as a compliment to `keyd`, for people who use the Scandinavian keyboard. 

Could you provide some more information on how this would be used with `keyd` for a Scandinavian keyboard? Could you also prove a link to the project? A quick search doesn't provide anything concrete for xkbmap and just returns xorg related tools which I would not expect to be useful on a system that doesn't use Xorg.

**Comment by gustaphe** [comment]
Okay I'm not very good at display manager level things.
 
[The keyd docs](https://github.com/rvaiya/keyd/blob/85dc69b57c38d432147a73940fc28252450fdcac/docs/keyd.scdoc?plain=1#L902C1-L904C55) say to use `setxkbmap -option compose:menu` to enable the compose-hack that allows typing exötic letters. From the documentation there, `xkbmap` appears to be something of a requirement. But I guess I got caught in an XY problem.

**Comment by Eeems** [comment]
> Okay I'm not very good at display manager level things.
> 
> [The keyd docs](https://github.com/rvaiya/keyd/blob/85dc69b57c38d432147a73940fc28252450fdcac/docs/keyd.scdoc?plain=1#L902C1-L904C55) say to use `setxkbmap -option compose:menu` to enable the compose-hack that allows typing exötic letters. From the documentation there, `xkbmap` appears to be something of a requirement. But I guess I got caught in an XY problem.

Right, so there is no display server running on the rM. It would also require the keyboard handling code that xochitl uses to support compose mode, which from what I can see in the [code that is public](https://github.com/reMarkable/epaper-qpa), it does not.

**Comment by gustaphe** [comment]
Xochitl does *something* to interpret `;` as `ö`. But okay, I guess it's not this simple. Will investigate.

**Comment by palfrey** [comment]
Regenesis is a bit broken, see https://github.com/irth/regenesis/issues/8

**Comment by palfrey** [comment]
https://github.com/toltec-dev/toltec/pull/957 is an early hack towards syncthingctl, but it's a _hard_ build. See notes there.
